Automattic/jetpack
# Add pre commit hook for linting JS

Fixes the surprise when looking Travis fail due to JS linting errors.

#### Testing instructions

1. checkout to this branch.
2. Run `yarn` so new dependencies are installed
3. Introduce a local change in a file (e.g. `_inc/client/main.jsx` ) adding a lint problem. (An unused var or a `let` that should be `const`)
4. Attempt to commit the change.
5. Expect the commit to fail.
6. Fix the error and add a change that does not present problems (an extra space between imports or so).
7. Commit the change and expect to not be blocked by the linter.

#### Changes proposed in this Pull Request:

* Upgrades all eslint rules to errors expect for `react/prefer-es6-class` and `react/no-bind`. Those remain as wranings.
* Adds `no-unused-vars` rule`
* Adds `husky` as dependency.
* Makes git ignore `.eslintcache`.
* Adds package.json script `precommit` to run `bin/pre-commit-hook.js`
